pkg/invariants/availability_image_registry/invariant.go

```diff
@@ -0,0 +1,212 @@
+package availability_image_registry
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/openshift/origin/pkg/invariants"
+	"github.com/openshift/origin/pkg/monitor/backenddisruption"
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"github.com/openshift/origin/test/extended/util/disruption"
+	"github.com/openshift/origin/test/extended/util/imageregistryutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	newConnectionTestName    = "[sig-imageregistry] Image registry remains available using new connections"
+	reusedConnectionTestName = "[sig-imageregistry] Image registry remains available using reused connections"
+)
+
+type availability struct {
+	kubeClient         kubernetes.Interface
+	routeClient        routeclient.Interface
+	imageRegistryRoute *routev1.Route
+
+	newConnectionDisruptionSampler    *backenddisruption.BackendSampler
+	reusedConnectionDisruptionSampler *backenddisruption.BackendSampler
+
+	// TODO stop storing this and use clients so we can unit test if we want
+	adminRESTConfig *rest.Config
+}
+
+func NewAvailabilityInvariant() invariants.InvariantTest {
+	return &availability{}
+}
+
+func (w *availability) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	var err error
+
+	w.adminRESTConfig = adminRESTConfig
+	w.kubeClient, err = kubernetes.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+	w.routeClient, err = routeclient.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+
+	w.imageRegistryRoute, err = imageregistryutil.ExposeImageRegistryGenerateName(ctx, w.routeClient, "test-disruption-")
```

Comment:
This might be where an invalid route is being created. Need to drop the dashes I think?

Reply:
it was setting both name and prefix.  fixed in updated in #28104